lib/modules/platform/github/index.ts

```diff
@@ -1318,12 +1334,26 @@ export async function ensureCommentRemoval(
 
 // Pull Request
 
-async function tryPrAutomerge(
+export async function tryPrAutomerge(
   prNumber: number,
   prNodeId: string,
-  platformOptions: PlatformPrOptions | undefined
+  platformOptions: PlatformPrOptions | undefined,
+  platformConfig: PlatformConfig
 ): Promise<void> {
-  if (platformConfig.isGhe || !platformOptions?.usePlatformAutomerge) {
+  // If GitHub Enterprise Server <3.3.0 it doesn't support automerge
+  if (platformConfig.isGhe) {
+    // semver not null safe, accepts null and undefined
+    // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
+    if (semver.satisfies(platformConfig.gheVersion!, '<3.3.0')) {
+      logger.debug(
+        { prNumber },
+        'GitHub-native automerge: not supported on this GHE version. Requires >=3.3.0'
+      );
+      return;
+    }
+  }
+
+  if (!platformOptions?.usePlatformAutomerge) {
     return;
   }
```

Comment:
I think this if should come first, and the GHE if second. If someone's on GHE 3.2.x but hasn't tried enabling platform automerge, there's no point to confuse them with a debug message saying it's not supported.